src/srs.jl

```diff
@@ -0,0 +1,174 @@
+    key_reversion::UInt16
+    minor_revision::UInt16
+    number_of_keys::UInt16
+    keys::Vector{KeyEntry}
+end
+
+# version numbers are fixed in the LAS specification
+GeoKeys(keys::Vector{KeyEntry}) = GeoKeys(0x0001, 0x0001, 0x0000, UInt16(length(keys)), keys)
+
+struct GeoDoubleParamsTag
+    double_params::Vector{Float64}
+end
+
+struct GeoAsciiParamsTag
+    ascii_params::String
+    nb::Int  # number of bytes
+    GeoAsciiParamsTag(s::AbstractString, nb::Integer) = new(ascii(s), Int(nb))
+end
+
+const id_geokeydirectorytag = UInt16(34735)
+const id_geodoubleparamstag = UInt16(34736)
+const id_geoasciiparamstag = UInt16(34737)
+
+"test whether a vlr is a GeoKeyDirectoryTag, GeoDoubleParamsTag or GeoAsciiParamsTag"
+is_srs(vlr::LasVariableLengthRecord) = vlr.record_id in (
+    id_geokeydirectorytag,
+    id_geodoubleparamstag,
+    id_geoasciiparamstag)
+
+# length in number of bytes
+Base.length(data::GeoKeys) = 8 * Int(data.number_of_keys) + 8
+Base.length(data::GeoDoubleParamsTag) = length(data.double_params) * 8
+Base.length(data::GeoAsciiParamsTag) = data.nb
+
+"Construct a projection VLR based on an EPSG code"
+function LasVariableLengthRecord(header::LasHeader, epsg::Integer)
+    if is_wkt(header)
+        throw(ArgumentError("Setting the SRS in WKT format not implemented"))
+    end
+
+    reserved = 0xAABB
+    user_id = "LASF_Projection"
+    description = "GeoTIFF GeoKeyDirectoryTag"
+    record_id = id_geokeydirectorytag
+
+    data = GeoKeys(epsg)
+
+    return LasVariableLengthRecord(
+        reserved,
+        user_id,
+        record_id,
+        description,
+        data
+    )
+end
+
+function Base.write(io::IO, data::GeoKeys)
+    write(io, data.key_directory_version)
+    write(io, data.key_reversion)
+    write(io, data.minor_revision)
+    write(io, data.number_of_keys)
+    for keyEntry in data.keys
+        write_key_entry(io, keyEntry)
+    end
+end
+
+Base.write(io::IO, data::GeoDoubleParamsTag) = write(io, data.double_params)
+Base.write(io::IO, data::GeoAsciiParamsTag) = writestring(io, data.ascii_params, data.nb)
+
+"Create GeoKeys from EPSG code. Assumes CRS is projected and in meters."
+function GeoKeys(epsg::Integer)
+    #Standard types
+    is_projected = KeyEntry(UInt16(1024), UInt16(0), UInt16(1), UInt16(1))         # Projected
+    proj_linear_units = KeyEntry(UInt16(1025), UInt16(0), UInt16(1), UInt16(1))    # Units in meter
+    projected_cs_type = KeyEntry(UInt16(3072), UInt16(0), UInt16(1), UInt16(epsg)) # EPSG code
+    vertical_units = KeyEntry(UInt16(3076), UInt16(0), UInt16(1), UInt16(9001))    # Units in meter
+    keys = [is_projected, proj_linear_units, projected_cs_type, vertical_units]
+    GeoKeys(keys)
+end
+
+function write_key_entry(io, entry::KeyEntry)
+    write(io, entry.keyid)
+    write(io, entry.tifftaglocation)
+    write(io, entry.count)
+    write(io, entry.value_offset)
+end
+
+"Get the EPSG code of the projection in the header"
+function epsg(header::LasHeader)
```

Comment:
Perhaps `epsg_code` and `epsg_code!` for these function names?

I'm also thinking a little about how this about would generalize to WKT support (which is oh so much nicer IMHO).  In that case we may want functions with names which are a little more general, eg `crs!` or `coordinate_reference_system!` or something.  But that seems like it might be over generalization until there's more julia ecosystem support for CRS metadata.